src/main/java/AutoResponder.java

```diff
@@ -0,0 +1,238 @@
+            Matcher m = PATTERN_TODO.matcher(input);
+            m.find();
+            return this.addTodo(m.group(1));
+        } else if (PATTERN_BYE.matcher(input).find()) {
+            System.exit(0);
+            return "";
+        } else {
+            throw new IllegalArgumentException("OOPS!!! I'm sorry, but I don't know what that means :-(");
+        }
+    }
+
+    private String processList() {
+        for (int i = 1; i <= taskList.size(); ++i) {
+            toPrint.append(i).append(". ").append(taskList.get(i - 1)).append("\n");
+        }
+        return printToConsole();
+    }
+
+    private String saveList() {
+        StringBuilder writeContents = new StringBuilder();
+        for (Task v : taskList) {
+            writeContents.append(v.writeFormat());
+            writeContents.append("\n");
+        }
+        if (!Files.exists(Paths.get("./data"))) {
+            boolean ok = new File("./data").mkdir();
+            if (!ok) {
+                System.err.println("Error in creating directory.\n");
+            }
+        }
+        File f = new File("./data/duke.txt");
+        String path = f.getAbsolutePath();
+        try {
+            FileWriter fw = new FileWriter(path);
+            fw.write(writeContents.toString());
+            fw.close();
+            toPrint.append("Tasks saved successfully.\n");
+        } catch (IOException e) {
+            toPrint.append("Tasks not loaded");
+        }
+        return this.printToConsole();
+    }
+
+    private String loadList() {
+        try {
+            File f = new File("./data/duke.txt");
+            Scanner sc = new Scanner(f);
+            String s;
+            while (sc.hasNextLine()) {
+                s = sc.nextLine();
+                switch (s.charAt(0)) {
+                case 'T':
+                    taskList.add(Todo.readFormat(s));
+                    break;
+                case 'D':
+                    taskList.add(Deadline.readFormat(s));
+                    break;
+                case 'E':
+                    taskList.add(Event.readFormat(s));
+                    break;
+                default:
+                    throw new IllegalArgumentException("No corresponding command found");
+                }
+            }
+            toPrint.append("Task file loaded successfully!\n");
+        } catch (Exception e) {
+            toPrint.append("Task file not loaded. Check if file exists or is corrupted.\n");
+        }
+        return this.printToConsole();
+    }
+
+    private String markTaskDone(int index) {
+        if (index < 0 || index >= taskList.size()) {
+            throw new IndexOutOfBoundsException("Index of " + (index + 1)
+                    + " does not correspond to task list of size " + taskList.size());
+        } else {
+            taskList.get(index).makeCompleted();
+            toPrint.append("Nice! I've marked this task as done:\n\t");
+            toPrint.append(taskList.get(index)).append("\n");
```

Comment:
Will it change the complete status of the task object in tasklist? I think your makeComplete() method only returns a new task object instead right?

Reply:
Thanks for spotting this error!

Reply:
Got it. Thanks for the clarification!